pkg/kubectl/cmd/create.go

```diff
@@ -201,8 +201,10 @@ func NameFromCommandArgs(cmd *cobra.Command, args []string) (string, error) {
 
 // CreateSubcommandOptions is an options struct to support create subcommands
 type CreateSubcommandOptions struct {
-	// Name of resource being created
+	// Name of resource instance being created (ex: my-pod, my-namespace)
 	Name string
+	// Resource being created (ex: pod, namespace)
+	Resource string
```

Comment:
It knows whats the resource its creating.
For ex: `kubectl create namespace my-namespace`

Mapper is the central place that knows the appropriate gvk for each resource.
We will be duplicating that information if I pass gvk here.